components/core/src/package/install.rs

```diff
@@ -50,6 +50,22 @@ impl PackageInstall {
         }
     }
 
+    /// Verifies an installation of a package that is equal or newer to a given ident and returns
+    /// a Result of a `PackageIdent` if one exists.
+    ///
+    /// An optional `fs_root` path may be provided to search for a package that is mounted on a
+    /// filesystem not currently rooted at `/`.
+    pub fn load_at_least(ident: &PackageIdent,
+                         fs_root_path: Option<&Path>)
+                         -> Result<PackageInstall> {
+        let package_install = try!(Self::resolve_package_install_min(ident, fs_root_path));
+        let package_target = try!(package_install.target());
+        match package_target.validate() {
```

Comment:
Since you're not using the inner type of a successful result of this call, it's probably better represented as

```
try!(package_target.validate());
Ok(package_install)
```